### DISCUSSION TITLE
Help |  error loading model: unexpectedly reached end of file

### DISCUSSION THREAD

**Discussion by kpshukla** [ggml-org/llama.cpp#1715] [Q&A] [answered]
Running llama.cpp on intel MacBook Pro.  I get the following error while running
./examples/alpaca.sh

I saved the model file in ./models

Error:
llama.cpp: loading model from ./models/ggml-alpaca-7b-q4.bin
error loading model: unexpectedly reached end of file
llama_init_from_file: failed to load model
llama_init_from_gpt_params: error: failed to load model './models/ggml-alpaca-7b-q4.bin'
main: error: unable to load model

What could be the problem?

**Comment by jyviko** [comment]
Same here:
```
main: build = 635 (5c64a09)
main: seed  = 1686146476
llama.cpp: loading model from ./models/7B/ggml-model-q4_0.bin
error loading model: unexpectedly reached end of file
llama_init_from_file: failed to load model
llama_init_from_gpt_params: error: failed to load model './models/7B/ggml-model-q4_0.bin'
main: error: unable to load model
```
System: MacBookPro15,2, Intel Core i7

**Comment by jspenaq** [comment]
Same error (MacBook M1 Pro, Ventura 13.3):

```
 ./main -m ./models/7B/ggml-model-q4_0.bin -p "Building a website can be done in 10 simple steps:" -n 512
main: build = 635 (5c64a09)
main: seed  = 1686150489
llama.cpp: loading model from ./models/7B/ggml-model-q4_0.bin
error loading model: unexpectedly reached end of file
llama_init_from_file: failed to load model
llama_init_from_gpt_params: error: failed to load model './models/7B/ggml-model-q4_0.bin'
main: error: unable to load model
```

**Reply by jspenaq** [reply]
I fix it by deleting `ggml-model-f16.bin` and `ggml-model-q4_0.bin`, then convert and quantize again.

**Reply by snwfdhmp** [reply]
can you provide a bash script ?

**Reply by kpshukla** [reply]
I am still getting the error...replaced the bin file

llama.cpp: loading model from ./models/ggml-alpaca-7b-q4.bin
error loading model: unexpectedly reached end of file
llama_init_from_file: failed to load model
llama_init_from_gpt_params: error: failed to load model './models/ggml-alpaca-7b-q4.bin'
main: error: unable to load model

**Reply by jyviko** [reply, marked as answer]
@kpshukla I think it's the model. Quantized my own model and it worked

**Reply by bolorundurovj** [reply]
@jyviko how'd you quantize the model?

**Reply by xelibrion** [reply]
@bolorundurovj get LLaMA weights and follow instructions here https://github.com/tatsu-lab/stanford_alpaca to get a complete representation of Alpaca model. Then follow instructions from README of this repo to quantize the model (it's written for LLaMA, but works for Alpaca as well).
